File: p2.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ConsoleApp1
{
    class p2
    {
        public static void Main(String[] args)
        {
            int total = 2;
            int a = 1;
            int b = 2; 
            while(b < 4000000)
            {
                int c = a + b;
                if (c%2 == 0) { total += c; }
                a = b;
                b = c; 
            }
            Console.WriteLine(total);
        }
                
    }
}

File: p2.py
import time
s = time.time()
f = [0,1]
total = 0
while f[-1] < 4000000:
    f2 = f[0] + f[1]
    f[0] = f[1]
    f[1] = f2
    if f[1] % 2 == 0: total += f[1]

print(total, time.time() - s)
